ch_pipeline/analysis/flagging.py

```diff
@@ -2260,3 +2261,195 @@ def process(self, data):
+    """Flag times following periods of heavy rainfall.
+
+    This task uses rainfall measurements from the DRAO weather station to compute
+    the accumulated rainfall within a given time interval prior to each time sample.
+    If the rainfall total exceeds some threshold, the weight dataset is set to zero
+    for that time.
+
+    Parameters
+    ----------
+    accumulation_time : float
+        Number of hours over which to compute accumulated rainfall for each time sample.
+        Default: 30.
+    threshold : float
+        Rainfall threshold (in mm) for flagging. Default: 1.0.
+    node_spoof : dictionary
+        Host and directory for finding weather data.
+        Default: {'cedar_online': '/project/rpp-krs/chime/chime_online/'}
+    """
+
+    accumulation_time = config.Property(proptype=float, default=30.0)
+    threshold = config.Property(proptype=float, default=1.0)
+    node_spoof = config.Property(proptype=dict, default=_DEFAULT_NODE_SPOOF)
+
+    def process(self, timestream):
+        """Set weight to zero if cumulative rainfall exceeds desired threshold.
+
+        Parameters
+        ----------
+        timestream : andata.CorrData or dcontainers.SiderealStream or dcontainers.TimeStream
+            Timestream to flag.
+
+        Returns
+        -------
+        timestream : andata.CorrData or dcontainers.SiderealStream or dcontainers.TimeStream
+            Returns the same timestream object with a modified weight dataset.
+        """
+        # Redistribute over the frequency direction
+        timestream.redistribute("freq")
+
+        # Get time axis or convert RA axis to Unix time
+        if "ra" in timestream.index_map:
+            ra = timestream.index_map["ra"][:]
+            if "lsd" in timestream.attrs:
+                csd = timestream.attrs["lsd"]
+            else:
+                csd = timestream.attrs["csd"]
+            time = ephemeris.csd_to_unix(csd + ra / 360.0)
+        else:
+            time = timestream.time
+
+        # Compute cumulative rainfall within specified time interval.
+        # Only run on rank 0, because a database query is required
+        if self.comm.rank == 0:
+            rainfall = compute_cumulative_rainfall(
+                time,
+                accumulation_time=self.accumulation_time,
+                node_spoof=self.node_spoof,
+            )
+        else:
+            rainfall = np.empty_like(time)
+
+        # Broadcast cumulative rainfall to all ranks
+        self.comm.Bcast(rainfall, root=0)
+
+        # Compute mask corresponding to times when rainfall is below threshold
+        rainfall_mask = rainfall < self.threshold
+
+        # Multiply weights by mask
+        weight = timestream.weight[:]
+        weight.local_array[:] *= rainfall_mask[np.newaxis, np.newaxis, :]
+
+        # Report how much data has been flagged due to rainfall
+        self.log.info(
+            "%0.2f percent of data was flagged due to rainfall."
```

Comment:
```suggestion
            f"{100.0 * (1.0 - np.sum(rainfall_mask) / Len(rainfall_mask):.2f} percent of data was flagged due to rainfall."
```
Linter isn't enforced in `ch_pipeline` GitHub CI yet, but it will end up complaining about older-style strings eventually